index.js

```diff
@@ -41,9 +41,24 @@ module.exports = function * (next) {
     return typeof item == 'number';
   });
 
+  //Adjust infinite end
+  if(end == Number.POSITIVE_INFINITY) {
+    if(this.length) {
+      end = this.length-1;
+    } else {
+      // This is ugly but we have to set something
+      end = 10000000000;
```

Comment:
I think that is overkill. The browser might act strange. The problem is that this scenario isn't really allowed by the spec. You must specify how much you are sending so this is something of a hack. The other alternative is to error out.